abi_stable/src/erased_types/trait_objects.rs

```diff
@@ -21,7 +21,21 @@ use crate::{
 #[derive(StableAbi)]
 pub struct HasherObject<'a> {
     this: RMut<'a, ErasedObject>,
-    hash_slice: unsafe extern "C" fn(RMut<'_, ErasedObject>, RSlice<'_, u8>),
+    write: unsafe extern "C" fn(RMut<'_, ErasedObject>, RSlice<'_, u8>),
```

Comment:
These write fields should move to a `&'static WriteFns` field with how many there are.

My general idea for the `WriteFns` struct
```rust
struct WriteFns{
    write: unsafe extern "C" fn(RMut<'_, ErasedObject>, RSlice<'_, u8>),
    write_i16: unsafe extern "C" fn(RMut<'_, ErasedObject>, i16),
    .... and all the other write_* function pointers
}

struct MakeWriteFns<T>;

impl<T> MakeWriteFns<T> 
where
    T: Hasher
{
    const V: &'static WriteFns = &WriteFns {
        write: write_Hasher::<T>,
        ....
    };
}
```

Reply:
Yeah, that definitely looks tidier. Let me know if the last commit is what you expected.